config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +113,199 @@ const (
+
+	// customConfig must be set when deploymentMode is CustomConfig, and must be unset otherwise.
+	// When set to CustomConfig, the Alertmanager will be deployed with custom configuration.
+	// +optional
+	CustomConfig *AlertmanagerCustomConfig `json:"customConfig,omitempty"`
+}
+
+// AlertmanagerDisabledConfig represents the configuration when Alertmanager is disabled.
+type AlertmanagerDisabledConfig struct {
+}
+
+// AlertmanagerDefaultConfig represents the configuration when using default Alertmanager settings.
+type AlertmanagerDefaultConfig struct {
+}
+
+// AlertmanagerCustomConfig represents the configuration for a custom Alertmanager deployment.
+// alertmanagerCustomConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+// +kubebuilder:validation:MinProperties=1
+type AlertmanagerCustomConfig struct {
+	// logLevel defines the verbosity of logs emitted by Alertmanager.
+	// This field allows users to control the amount and severity of logs generated, which can be useful
+	// for debugging issues or reducing noise in production environments.
+	// Allowed values are Error, Warn, Info, Debug, and omitted.
+	// When set to Error, only errors will be logged.
+	// When set to Warn, both warnings and errors will be logged.
+	// When set to Info, general information, warnings, and errors will all be logged.
+	// When set to Debug, detailed debugging information will be logged.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, that is subject to change over time.
+	// The current default value is `Info`.
+	// +optional
+	LogLevel LogLevel `json:"logLevel,omitempty"`
+	// nodeSelector is the node selector applied to network diagnostics components
+	// nodeSelector is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// The current default value is `kubernetes.io/os: linux`.
+	// +optional
+	// +kubebuilder:validation:MinProperties=1
+	// +kubebuilder:validation:MaxProperties=10
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// resources defines the compute resource requests and limits for the Alertmanager container.
+	// This includes CPU, memory and HugePages constraints to help control scheduling and resource usage.
+	// When not specified, defaults are used by the platform. Requests cannot exceed limits.
+	// This field is optional.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// This is a simplified API that maps to Kubernetes ResourceRequirements.
+	// The current default values are:
+	//   resources:
+	//     requests:
+	//       cpu: 4m
+	//       memory: 40Mi
```

Comment:
This format is not a correct example of the actual format for the field. Instead, it should be something like:
```
resources:
  - name: cpu
    request: 4m
    limit: 5m
```